### PR TITLE
Refactor fetchTweet method to async await

### LINKED ISSUE #141: REFACTOR - Async / Await
## Describe the feature you want the app to have that it currently doesn't
* Refactor and use async / await syntax for fetching a tweet

## Describe alternatives you've considered
* N/A

## Additional context
* In `AddURLViewModel.swift` in the function `fetchTweet` change the function to use async / await syntax

### PR DESCRIPTION
**Pull request by Drag0ndust**
## What it Does
* Refactor the `fetchTweet` method to use async/await instead of a completion handler

## Additional refactoring
* Think about creating an issue to refactor the methods in `DataStoreManger` to async/await

Closes #141

### CODE REVIEW

**Comment by Drag0ndust** [comment]
@mikaelacaron Now we start new with this PR, hope now it is ok ;)

File: brain-marks/Add/AddURLView.swift [hunk 1]
```diff
@@ -40,20 +40,20 @@ struct AddURLView: View {
                             showingAlert = true
                             
                         } else {
-                            viewModel.fetchTweet(url: newEntry) { result in
-                                switch result {
-                                case .success(let tweet):
+                            Task(priority: .userInitiated) {
+                                do {
+                                    let tweet = try await self.viewModel.fetchTweet(url: newEntry)
```
Comment (mikaelacaron) [inline brain-marks/Add/AddURLView.swift]:
```suggestion
                                    let tweet = try await viewModel.fetchTweet(url: newEntry)
```

File: brain-marks/Add/AddURLView.swift [hunk 2]
```diff
@@ -40,20 +40,20 @@ struct AddURLView: View {
...
                                                 tweet: tweet,
                                                 category: selectedCategory)
                                         }
                                         presentationMode.wrappedValue.dismiss()
-                                        
                                     }
-                                case .failure:
-                                    viewModel.alertItem = AlertContext.badURL
+                                } catch {
+                                    self.viewModel.alertItem = AlertContext.badURL
```
Comment (mikaelacaron) [inline brain-marks/Add/AddURLView.swift]:
```suggestion
                                   viewModel.alertItem = AlertContext.badURL
```

File: brain-marks/Add/AddURLViewModel.swift [hunk 3]
```diff
@@ -17,47 +17,24 @@ enum HttpError: Error {
...
+
+        if let response = response as? HTTPURLResponse {
+            guard (200 ... 299) ~= response.statusCode else {
+                Logger.network.error("❌ Status code is \(response.statusCode)")
+                throw HttpError.badResponse
+            }
+
+            do {
+                let result = try JSONDecoder().decode(Response.self, from: data)
+                return self.createTweet(result)
```
Comment (mikaelacaron) [inline brain-marks/Add/AddURLViewModel.swift]:
```suggestion
                return createTweet(result)
```

**Review by mikaelacaron — APPROVED** [review]
Good work!! We didn't need `self` in several places, I'll remove them and merge this!

**Pull request merged**